delegator/src/sign/sign.service.ts

```diff
@@ -97,26 +93,24 @@ export class SignService {
     return tx.getRawTransaction()
   }
 
-  validateTransaction(tx) {
-    // FIXME remove simple whiteListing settings and setup db
-    const contractList = {
-      '0x93120927379723583c7a0dd2236fcb255e96949f': {
-        methods: ['increment()', 'decrement()'],
-        reporters: ['0x260836ac4f046b6887bbe16b322e7f1e5f9a0452']
+  async validateTransaction(tx) {
+    const result = await this.prisma.contract.findMany({
+      where: {
+        address: tx.to,
+        Reporter: {
+          some: {
+            address: tx.from
+          }
+        },
+        Function: {
+          some: {
+            encodedName: tx.input.substring(0, 11)
```

Comment:
I think it should be length of 10.

```suggestion
            encodedName: tx.input.substring(0, 10)
```